lib/hls/hls_parser.js

```diff
@@ -3616,6 +3624,130 @@ shaka.hls.HlsParser = class {
     return [startByte, endByte];
   }
 
+  /**
+   * @param {!Array.<!shaka.hls.Tag>} tags
+   * @private
+   */
+  processDateRangeTags_(tags) {
```

Comment:
If you pass in the `type`, too, then you could use the `ignoreManifestProgramDateTimeFor_ ` helper function, instead of just checking `this.config_.hls.ignoreManifestProgramDateTime`. That will also let this account for the `ignoreManifestProgramDateTimeForTypes` configuration.

Reply:
Done!